lib/utils/transform.py

```diff
@@ -121,3 +128,34 @@ def changecase(index, word):
             return word.title()
 
     return ''.join(changecase(i, w) for i, w in words)
+
+
+def format_date(date_string: str, expected_formats: set) -> str:
+    """
+    Format *date_string* to ISO 8601 date (YYYY-MM-DD).
+    If *date_string* does not match *expected_formats*, return *date_string*.
+
+    >>> expected_formats = {'%Y-%m-%d', '%Y-%m-%dT%H:%M:%SZ'}
+
+    >>> format_date("2020", expected_formats)
+    '2020'
+
+    >>> format_date("2020-01", expected_formats)
+    '2020-01'
+
+    >>> format_date("2020-1-15", expected_formats)
+    '2020-01-15'
+
+    >>> format_date("2020-01-15", expected_formats)
+    '2020-01-15'
+
+    >>> format_date("2020-01-15T00:00:00Z", expected_formats)
+    '2020-01-15'
+    """
+    for format in expected_formats:
```

Comment:
`format` is a built-in function in python, so it might be better to choose a different variable name here.

Reply:
Ah, thanks. Updated in latest commit.